src/index.js

```diff
@@ -57,7 +57,13 @@ export default class Router extends BaseRouter {
           // TODO(@max) sort here???
           async.eachSeries(
             _.map(denyResults, 'middleware'),
-            (middleware, callback) => middleware(req, res, callback),
+            (middleware, callback) => {
+              try {
+                middleware(req, res, callback);
+              } catch (err) {
```

Comment:
To aid in future debugging, it might be good to add a `debug` statement here:

```js
debug('DENY middleware threw a synchronous error', err.stack);
```